lib/ansible/modules/monitoring/zabbix/zabbix_host.py

```diff
@@ -780,6 +780,8 @@ def main():
                 interface['ip'] = ''
             if 'main' not in interface:
                 interface['main'] = 0
+            if 'port' in interface and not isinstance(interface['port'], str):
+                module.fail_json(msg="port should be an string on interface '%s'." % interface)
```

Comment:
I wouldn't go as far as failing module because someone has not provided integer in quotes. This would require us to deprecate the feature first and then remove it in usually 4th release from now (that would be ansible 2.14)

What about this instead?

```suggestion
                try:
                    interface['port'] = str(interface['port'])
                except ValueError:
                    module.fail_json(msg="port should be convertable to string on interface '%s'." % interface)
```

Reply:
Way better!